tests/admin_changelist/tests.py

```diff
@@ -1995,3 +2069,96 @@ def test_collapse_filter_with_unescaped_title(self):
                 By.CSS_SELECTOR, "[data-filter-title='It\\'s OK']"
             ).get_attribute("open")
         )
+
+    def test_list_display_ordering(self):
+        from selenium.webdriver.common.by import By
+
+        parent_a = Parent.objects.create(name="Parent A")
+        parent_b = Parent.objects.create(name="Parent B")
+        parent_c = Parent.objects.create(name="Parent C")
+        child_l = Child.objects.create(name="Child L", parent=parent_b)
+        child_m = Child.objects.create(name="Child M", parent=parent_a)
+        child_n = Child.objects.create(name="Child N", parent=parent_c)
+        GrandChild.objects.create(name="Grandchild X", parent=child_m)
+        GrandChild.objects.create(name="Grandchild Y", parent=child_l)
+        GrandChild.objects.create(name="Grandchild Z", parent=child_n)
```

Comment:
Just to cover our basis, could we have a parent and/or child and/or grandchild with `parent=None`? (I know that conceptually it feels weird, but since the models allow it and it caused bugs in previous revisions, it may be worth it?)